src/main/scala/beam/agentsim/agents/ridehail/RideHailManager.scala

```diff
@@ -1328,4 +1370,13 @@ class RideHailManager(
     rideInitialLocation
   }
 
+  private def convertToShiftString(startTimes: ArrayBuffer[Int], endTimes: ArrayBuffer[Int]): Option[String] = {
+    if (startTimes.length != endTimes.length) {
+      None
+    } else {
+      var outStr: String = new String
+      (startTimes zip endTimes).foreach(x => outStr += ("{" + x._1.toString + ":" + x._2.toString + "}"))
```

Comment:
This should be a stringbuilder at minimum to avoid garbage. But in this case you could probably turn it into a `mkString`